drivers/dma/dma_esp32_gdma.c

```diff
@@ -157,17 +162,42 @@ static int dma_esp32_disable_interrupt(const struct device *dev,
 
 #endif
 static int dma_esp32_config_rx_descriptor(struct dma_esp32_channel *dma_channel,
-					   struct dma_block_config *block)
+						struct dma_block_config *block)
 {
+	if (!block) {
+		LOG_ERR("At least one dma block is required");
+		return -EINVAL;
+	}
+
 	if (!esp_ptr_dma_capable((uint32_t *)block->dest_address)) {
 		LOG_ERR("Rx buffer not in DMA capable memory: %p", (uint32_t *)block->dest_address);
 		return -EINVAL;
 	}
 
-	memset(&dma_channel->desc, 0, sizeof(dma_channel->desc));
-	dma_channel->desc.buffer = (void *)block->dest_address;
-	dma_channel->desc.dw0.size = block->block_size;
-	dma_channel->desc.dw0.owner = DMA_DESCRIPTOR_BUFFER_OWNER_DMA;
+	dma_descriptor_t *desc_iter = dma_channel->desc_list;
```

Comment:
Line spaces are fine to separate code and render it more viewable.
Suggestion to keep spacing and coding style throughout the code